tests/python/pants_test/engine/test_fs.py

```diff
@@ -597,6 +598,39 @@ def test_caches_downloads(self):
           82
         ))
 
+  def generate_original_digest(self) -> Digest:
+    content = b'dummy content'
+    input_files_content = InputFilesContent((
+      FileContent(path='a.txt', content=content),
+      FileContent(path='b.txt', content=content),
+      FileContent(path='c.txt', content=content),
+      FileContent(path='subdir/a.txt', content=content),
+      FileContent(path='subdir/b.txt', content=content),
+      FileContent(path='subdir2/a.txt', content=content),
+    ))
+    digest = self.request_single_product(Digest, input_files_content)
+    return digest
+
+  def test_empty_snapshot_subset(self) -> None:
+    ss = SnapshotSubset(directory_digest=self.generate_original_digest(),
+        include_files=(),
+        include_dirs=(),
+    )
+    subset_digest = self.request_single_product(Digest, ss)
+    assert subset_digest == EMPTY_DIRECTORY_DIGEST
+
+  def test_snapshot_subset_files_dirs(self) -> None:
+    ss = SnapshotSubset(directory_digest=self.generate_original_digest(),
+        include_files=("a.txt", "c.txt"),
+        include_dirs=('subdir2',),
+    )
+    subset_digest = self.request_single_product(Digest, ss)
+    result_files = self.request_single_product(FilesContent, subset_digest)
+    assert len(result_files.dependencies) == 3
+    assert any(fc.path == 'a.txt' for fc in result_files.dependencies)
+    assert any(fc.path == 'c.txt' for fc in result_files.dependencies)
+    assert any(fc.path == 'subdir2/a.txt' for fc in result_files.dependencies)
+
 
```

Comment:
Can you please add a test for what happens if you specify `include_files` and `include_dirs` that aren't in the original digest? From the Rust implementation, it looks like we won't error? I think that's the right thing to do?

Reply:
I'd lean towards erroring, rather than ignoring, but a test either way would be good :)

Reply:
Bump on this. Btw, you control the behavior for this edge case with `GlobMatchErrorBehavior` and `GlobExpansionConjunction`

Reply:
Erroring is probably better in some sense, but ignoring is easier, and I don't want to spend more time on this, so I'll encode the ignoring behavior in a test and we can revisit this in a future PR if that turns out to be a problem.